.github/workflows/default.yaml

```diff
@@ -0,0 +1,128 @@
+  # while excessively shallow clones break tests that check for past commits
+  # (e.g., to skip a particular test until a known bug is fixed) or generate
+  # diffs against them (e.g., for `git diff --check`). This arbitrary limit
+  # tries to balance the two concerns.
+  CHECKOUT_FETCH_DEPTH: 1001
+
+jobs:
+
+  functionality-tests:
+
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Install prerequisite packages
+        run: |
+          sudo apt-get --quiet=2 update
+          sudo apt-get --quiet=2 install libtool-bin
+
+      - name: Setup a nodejs environment
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+
+      - name: Checkout Squid sources
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: ${{ env.CHECKOUT_FETCH_DEPTH }}
+
+      - run: ./bootstrap.sh
+      - run: ./configure --with-openssl
+      - run: make -j2
+      - run: |
+          sudo make install
+          sudo chown -R nobody:nogroup /usr/local/squid
+
+      - run: ./test-suite/test-functionality.sh
+
+      # Squid logs are not readable to actions/upload-artifact below
+      - name: Prep test logs
+        if: success() || failure()
+        run: sudo chmod -R a+rX /usr/local/squid
+
+      - name: Publish test logs
+        if: success() || failure()
+        uses: actions/upload-artifact@v3
+        with:
+          name: test-logs
+          path: |
+            ${{ runner.temp }}/*.log
+            /usr/local/squid/var/logs/overlord/*.log
+
+  source-maintenance-tests:
+
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Install prerequisite packages
+        run: |
+          sudo apt-get --quiet=2 update
+          sudo apt-get --quiet=2 install astyle
+          sudo apt-get --quiet=2 install gperf
+          pip install \
+              --user \
+              --no-cache-dir \
+              --disable-pip-version-check \
+              --quiet \
+              --progress-bar off \
+              codespell==1.16 # TODO: Upgrade to codespell v2
+
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: ${{ env.CHECKOUT_FETCH_DEPTH }}
+
+      - run: ./test-suite/test-sources.sh
+
+  build-tests:
+
+    strategy:
+      fail-fast: true
+      matrix:
+        os: [ ubuntu-22.04, macos-11 ]
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+
+      - name: Install prerequisite Linux packages
+        if: runner.os == 'Linux'
+        run: |
+          sudo apt-get --quiet=2 update
+          sudo apt-get --quiet=2 install libtool-bin
+          sudo apt-get --quiet=2 install libcppunit-dev
+          sudo apt-get --quiet=2 install libldap2-dev
+
+      - name: Install prerequisite macOS packages
+        if: runner.os == 'macOS'
+        run: |
+           brew install automake
+           brew install cppunit
+
+      - name: Checkout sources
+        uses: actions/checkout@v3
+
+      - run: ./test-builds.sh
+        # XXX: Fix distclean cleanup on macOS instead
```

Comment:
I posted #1159 that fix other _known_ macOS build errors, but I did not know enough how to fix the last remaining one:

```
ERROR: files left in build directory after distclean:
./src/cf_gen.dSYM/Contents/Resources/DWARF/cf_gen
./src/cf_gen.dSYM/Contents/Info.plist
make[1]: *** [distcleancheck] Error 1
make: *** [distcheck] Error 1
buildtest.sh result is 2
Log end: btlayer-00-default.log
Error: Process completed with exit code 1.
```

If somebody can address it, please do so in a separate PR. FWIW, other projects had to fix similar-looking [bugs](https://code.monotone.ca/p/monotone/issues/view/attachment/39/Makefile.am.diff).

Reply:
MacOS is not one of the OS the Squid Foundation board has agreed on being considered "officially supported". It should not be required by SemaphoreCI at all until its build becomes bug-free and stable enough to join that set.

Reply:
see http://build.squid-cache.org/job/5-pr-test/ - CentOS, Debian, Fedora, OpenSuse, Ubuntu - latest three versions.

Reply:
That matrix list is the result of the recurring/regular discussions and analysis by the board members about which OS we are able to call "officially supported" as of the Squid-5 stable cycle.

Failure to build and test successfully on any of those listed OS blocks merge. Whereas issues seen on other environments **may** be ignored at reviewer discretion.

Reply:
> That matrix list is the result of the recurring/regular discussions and analysis by the board members
> 
> Failure to build and test successfully on any of those listed OS blocks merge. Whereas issues seen on other environments **may** be ignored at reviewer discretion.

I agree with the quoted parts. However, I do not think any of the above equates that Jenkins matrix with the "officially supported" designation (unless we have actually agreed to make that matrix equivalent to that designation, which would require, among other things, agreeing on never modifying that matrix without strong consensus). AFAICT, there is currently no agreement regarding the list of officially supported environments -- we still need to work that out. The lack of that agreement does not prevent Jenkins improvements and merging this PR, of course.

Reply:
That matrix is the unified sum of a series discussions and decisions about what OS can (or not) be officially supported for the Squid Software Foundation testing and promises of stability for Squid release series. Each release series has its own N-pr-test matrix relevant to the time(s) it was released.
```
  Grass does not need a field to grow - Nevyn
```

Reply:
> That matrix is the unified sum of a series discussions and decisions about what OS can (or not) be officially supported for the Squid Software Foundation testing and promises of stability for Squid release series.

I disagree that the Jenkins matrix is equivalent to the "officially supported" designation. Thus, there is currently no consensus that it is equivalent. It may become equivalent if we explicitly decide that it is or if the explicit past decision making it equivalent is discovered. Until then, there is no official definition of "officially supported".